kubernetes/spire/templates/agent/deployment.yaml

```diff
@@ -19,11 +19,22 @@ spec:
       labels:
         app.kubernetes.io/name: {{ template "spire.name" . }}-agent
     spec:
-      hostPID: true
       hostNetwork: true
       dnsPolicy: ClusterFirstWithHostNet
       serviceAccountName: {{ template "spire.name" . }}-agent
       initContainers:
+        - name: init-setperms
+          image: "{{ .Values.agent.init.repository }}:{{ .Values.agent.init.tag }}"
+          command:
+            - '/bin/sh'
+          args:
+            - '-x'
+            - '-c'
+            - 'chown -R 65534:65534 /run/spire/sockets'
+          volumeMounts:
+            - name: {{ template "spire.name" . }}-socket
+              mountPath: /run/spire/sockets
+              readOnly: false
         - name: init
           # This is a small image with wait-for-it, choose whatever image
```

Comment:
the image also has to have /bin/sh and chown now.
Same comment applies to kubernetes/spire/templates/server/deployment.yaml

Reply:
I'm going to switch it to use init2 instead. I'm kind of surprised the wait-for-it image worked with it. I'll run another test to validate init2 works fine and then close this comment.